src/main/java/org/codeforamerica/messaging/models/Template.java

```diff
@@ -20,33 +20,29 @@
 @AllArgsConstructor
 @NoArgsConstructor
 @Builder
+@Table(uniqueConstraints = {@UniqueConstraint(columnNames = {"name", "language", "variant"})})
```

Comment:
Unfortunately, this design will not allow for a message request in the general case (which has multiple recipients) to refer to a template unambiguously, ie, without a specific language or variant specified. This is why we had a 1:n relationship between template and variants in the prototype.

Reply:
I am imagining that the `MessageRequest.templateParams` could be used to pass in language and variant info per message, or otherwise use the default for that template

Reply:
Consider the basic scenario in the prototype where we request to send a `MessageBatch` with a template and a CSV of recipients. How would you associate the `MessageBatch` with a `Template` and persist it? In other words, which field in the `MessageBatch` is a foreign key to the `Template`?

Reply:
Does a `MessageBatch` need to be associated/persisted with more than a template name? I understand that a template name can refers to a group of templates, but right now I don't see the benefit of persisting a higher level template object to hold the 1:n relationship to templates. Would it be for versioning?

I am specifically imagining template names are case-insensitive with letters and underscores only, perhaps I should add that as a field pattern.

Reply:
The `Message` will have a unique `Template`, but in my mind a `MessageBatch` is associated to a list of `Templates`

Reply:
The MessageBatch will need to be associated with a single **template** but with the current design, there is no way for it to have the persisted data to enforce that relationship. What would the foreign key be on the `MessageBatch` table? 
> A foreign key must reference columns that either are a primary key or form a unique constraint.
-- https://www.postgresql.org/docs/14/ddl-constraints.html#DDL-CONSTRAINTS-FK

Reply:
Take a look at what I just added, I realized I hadn't fully implemented what I intended.